CMakeLists.txt

```diff
@@ -56,6 +56,10 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
         $<$<COMPILE_LANGUAGE:CXX>:-Wno-vla-extension>
         $<$<COMPILE_LANGUAGE:CXX>:-Wno-zero-length-array>
     )
+else()
```

Comment:
Is there something special about autodocs that triggers truncation warnings? I would have expected that truncation would be failing on normal gcc builds as part of CI.

I'm a little leary of conditionally disabling format truncation only if the compiler isn't clang, maybe always disable the format truncation warning if we're not ready to resolve it yet.

Reply:
The issue is that clang does not support this flag.  Thus we cannot disable it on clang.

Reply:
Gotcha. Maybe just add the flag if the compiler is GCC so we don't break in weird ways on unexpected compilers? That being said I think it's unlikely enough that somebody would build the root cmake project on an odd compiler, so maybe not a significant concern.